src/io/bnx.rs

```diff
@@ -0,0 +1,1486 @@
+                } else if buffer.starts_with("# Label Channels:\t2") {
+                    num_channels = 2;
+                } else {
+                };
+                header.push(String::from(buffer.trim_end()));
+            } else {
+                break;
+            }
+        }
+        for _ in 0..3 * num_channels {
+            stream.read_line(&mut buffer)?;
+        }
+        if num_channels == 0 {
+            bail!(Error::InvalidLabelChannel("1 or 2"))
+        } else {
+            Ok(Reader {
+                stream,
+                header,
+                num_channels,
+                buffer,
+            })
+        }
+    }
+}
+
+impl<R: BufRead> BnxRead for Reader<R> {
+    fn read_into(&mut self, record: &mut Record) -> Result<bool> {
+        if self.buffer.is_empty() {
+            return Ok(false);
+        }
+        let buf_lines: Vec<&str> = self.buffer.trim().lines().collect();
+        let mut res = match record.fill(buf_lines[0], buf_lines[1..].to_vec()) {
+            Ok(()) => Ok(true),
+            Err(e) => Err(e),
+        };
+        self.buffer.clear();
+        for _ in 0..3 * self.num_channels + 1 {
+            res = match self.stream.read_line(&mut self.buffer) {
+                Ok(_) => res,
+                Err(e) => Err(e.into()),
+            }
+        }
+        res
+    }
+}
+
+/// Iterator over records.
+impl<R: BufRead> Iterator for Reader<R> {
+    type Item = Result<Record>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let mut rec = Record::new();
+        match self.read_into(&mut rec) {
+            Ok(true) => Some(Ok(rec)),
+            Ok(false) => None,
+            Err(e) => Some(Err(e)),
+        }
+    }
+}
+
+/// A container with BNX header and records.
+#[derive(Debug, new, PartialEq)]
+pub struct Container {
+    header: Vec<String>,
+    molecules: HashMap<u32, Record>,
+}
+
+impl Container {
+    /// Constructs a new `Container` from path.
+    pub fn from_path<P: AsRef<Path>>(path: P) -> Result<Self> {
+        let reader = Reader::from_path(path)?;
+        reader.into_container()
+    }
+
+    /// Returns header.
+    pub fn header(&self) -> &Vec<String> {
+        &self.header
+    }
+
+    /// Returns label positions (for each molecule respectively).
+    pub fn labels(&mut self) -> HashMap<u32, (Vec<f64>, Vec<f64>)> {
+        self.molecules
+            .clone()
+            .iter_mut()
+            .map(|(id, rec)| {
+                (
+                    *id,
+                    (
+                        rec.channel_1()
+                            .iter()
+                            .map(|label| label.pos)
+                            .collect::<Vec<_>>(),
+                        rec.channel_2()
+                            .iter()
+                            .map(|label| label.pos)
+                            .collect::<Vec<_>>(),
+                    ),
+                )
+            })
+            .collect::<HashMap<_, _>>()
+    }
```

Comment:
Not the thing we actually need. Instead, one wants a method that returns the record object given an id.

Reply:
Why do we want the whole record? Wouldn't the label positions be more useful to us? 
Maybe add a function 'record()' and change the above to be a function of 'Record'?